Objects/typeobject.c

```diff
@@ -7000,6 +7005,34 @@ static slotdef slotdefs[] = {
     {NULL}
 };
 
+
+/* Since searching slotdefs[] is expensive and assigning to slots is rare,
+   do a quick probability check if the PyUnicode name could be a slot name
+   at all. */
+static int name_looks_like_slot(PyObject *name) {
+    Py_ssize_t length = PyUnicode_GET_LENGTH(name);
+    /* The longest slot name is currently "__getattribute__": 16 characters */
+    if (length >= 6 && length <= 16) {
```

Comment:
I don't like this. If in the future a new slot is added we need to remember to update this line and I am almost certain it will be missed.

Reply:
I was thinking that, too, that's why I implemented the function here and added the comment. But I can live with removing the maximum length check.